PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -2250,7 +2260,29 @@ void ScintillaEditView::restyleBuffer()
 
 void ScintillaEditView::styleChange()
 {
-	defineDocType(_currentBuffer->getLangType());
+	const bool isSameLangType = _prevBuffer != nullptr && ((_prevBuffer == _currentBuffer) || (_prevBuffer->getLangType() == _currentBuffer->getLangType()));
+	const int currentLangInt = static_cast<int>(_currentBuffer->getLangType());
+	const bool isFirstActiveBuffer = (_currentBuffer->getLastLangType() != currentLangInt) || (_currentBuffer->isUntitled());
+

```

Comment:
@softmgr 
Why `|| (_currentBuffer->isUntitled()` ?
If we copy all data from a large XML file, paste it to an untitled new document and set it to XML - switch off then switch back in it - there will be the performance issue.

Reply:
OK, I see the problem when `|| (_currentBuffer->isUntitled())` is removed.
So it should be kept.